librz/il/rzil_opcodes.c

```diff
@@ -655,128 +676,147 @@ RZ_API void rz_il_op_free(RZ_NULLABLE RzILOp *op) {
-		rz_il_op_free_1(neg, bv);
+		rz_il_op_free_1(pure, neg, bv);
 		break;
 	case RZIL_OP_LOGNOT:
-		rz_il_op_free_1(lognot, bv);
+		rz_il_op_free_1(pure, lognot, bv);
 		break;
 	case RZIL_OP_ADD:
-		rz_il_op_free_2(add, x, y);
+		rz_il_op_free_2(pure, add, x, y);
 		break;
 	case RZIL_OP_SUB:
-		rz_il_op_free_2(sub, x, y);
+		rz_il_op_free_2(pure, sub, x, y);
 		break;
 	case RZIL_OP_MUL:
-		rz_il_op_free_2(mul, x, y);
+		rz_il_op_free_2(pure, mul, x, y);
 		break;
 	case RZIL_OP_DIV:
-		rz_il_op_free_2(div, x, y);
+		rz_il_op_free_2(pure, div, x, y);
 		break;
 	case RZIL_OP_SDIV:
-		rz_il_op_free_2(sdiv, x, y);
+		rz_il_op_free_2(pure, sdiv, x, y);
 		break;
 	case RZIL_OP_MOD:
-		rz_il_op_free_2(mod, x, y);
+		rz_il_op_free_2(pure, mod, x, y);
 		break;
 	case RZIL_OP_SMOD:
-		rz_il_op_free_2(smod, x, y);
+		rz_il_op_free_2(pure, smod, x, y);
 		break;
 	case RZIL_OP_LOGAND:
-		rz_il_op_free_2(logand, x, y);
+		rz_il_op_free_2(pure, logand, x, y);
 		break;
 	case RZIL_OP_LOGOR:
-		rz_il_op_free_2(logor, x, y);
+		rz_il_op_free_2(pure, logor, x, y);
 		break;
 	case RZIL_OP_LOGXOR:
-		rz_il_op_free_2(logxor, x, y);
+		rz_il_op_free_2(pure, logxor, x, y);
 		break;
 	case RZIL_OP_SHIFTR:
-		rz_il_op_free_3(shiftr, fill_bit, x, y);
+		rz_il_op_free_3(pure, shiftr, fill_bit, x, y);
 		break;
 	case RZIL_OP_SHIFTL:
-		rz_il_op_free_3(shiftl, fill_bit, x, y);
+		rz_il_op_free_3(pure, shiftl, fill_bit, x, y);
+		break;
+	case RZIL_OP_EQ:
+		rz_il_op_free_2(pure, eq, x, y);
 		break;
 	case RZIL_OP_SLE:
-		rz_il_op_free_2(sle, x, y);
+		rz_il_op_free_2(pure, sle, x, y);
 		break;
 	case RZIL_OP_ULE:
-		rz_il_op_free_2(ule, x, y);
+		rz_il_op_free_2(pure, ule, x, y);
 		break;
 	case RZIL_OP_CAST:
-		rz_il_op_free_1(cast, val);
+		rz_il_op_free_1(pure, cast, val);
 		break;
 	case RZIL_OP_CONCAT:
 		rz_warn_if_reached();
 		break;
 	case RZIL_OP_APPEND:
-		rz_il_op_free_2(append, x, y);
+		rz_il_op_free_2(pure, append, x, y);
 		break;
 	case RZIL_OP_LOAD:
-		rz_il_op_free_1(load, key);
-		break;
-	case RZIL_OP_STORE:
-		rz_il_op_free_2(store, key, value);
+		rz_il_op_free_1(pure, load, key);
 		break;
 	case RZIL_OP_LOADW:
-		rz_il_op_free_1(loadw, key);
+		rz_il_op_free_1(pure, loadw, key);
+		break;
+	default:
+		rz_warn_if_reached();
+		RZ_LOG_ERROR("RzIl: unknown opcode %u\n", op->code);
+		break;
+	}
+	free(op);
+}
+
+RZ_API void rz_il_op_effect_free(RZ_NULLABLE RzILOpEffect *op) {
+	if (!op) {
+		return;
+	}
+	switch (op->code) {
+	case RZIL_OP_STORE:
+		rz_il_op_free_2(pure, store, key, value);
 		break;
 	case RZIL_OP_STOREW:
-		rz_il_op_free_2(storew, key, value);
+		rz_il_op_free_2(pure, storew, key, value);
 		break;
 	case RZIL_OP_NOP:
 		// nothing to free
 		break;
 	case RZIL_OP_SET:
-		rz_il_op_free_1(set, x);
+		rz_il_op_free_1(pure, set, x);
 		break;
 	case RZIL_OP_LET:
-		rz_il_op_free_1(let, x);
+		rz_il_op_free_1(pure, let, x);
 		break;
 	case RZIL_OP_JMP:
-		rz_il_op_free_1(jmp, dst);
+		rz_il_op_free_1(pure, jmp, dst);
```

Comment:
Is this on purpose¿

Reply:
I'm not sure what you are referring to. The `rz_il_op_free_1` macro receives a new argument to select the right functions for the type.

Reply:
`jmp->dst` is pure.